lib/rosette/integrations/smartling-integration/smartling_puller.rb

```diff
@@ -1,135 +1,217 @@
 # encoding: UTF-8
 
+require 'set'
+require 'concurrent'
+
 module Rosette
   module Integrations
     class SmartlingIntegration < Integration
       class SmartlingPuller
 
-        attr_reader :rosette_config, :smartling_apis, :completed_files_map, :locale
+        DEFAULT_THREAD_POOL_SIZE = 15
```

Comment:
Ya they're different operations. I think we cap our MySQL connections to 16, so I chose 15 to leave some wiggle room. I think however it might be better to use 10 here too to leave more than just a single free connection. What do you think?